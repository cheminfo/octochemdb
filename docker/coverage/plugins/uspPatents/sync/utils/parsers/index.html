
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for plugins/uspPatents/sync/utils/parsers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../../index.html">All files</a> plugins/uspPatents/sync/utils/parsers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">91.74% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>189/206</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">40% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>6/15</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>3/3</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">91.74% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>189/206</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="parseUsp2001To2004.js"><a href="parseUsp2001To2004.js.html">parseUsp2001To2004.js</a></td>
	<td data-value="96.61" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.61" class="pct high">96.61%</td>
	<td data-value="59" class="abs high">57/59</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="96.61" class="pct high">96.61%</td>
	<td data-value="59" class="abs high">57/59</td>
	</tr>

<tr>
	<td class="file high" data-value="parseUsp2005.js"><a href="parseUsp2005.js.html">parseUsp2005.js</a></td>
	<td data-value="97.01" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 97%"></div><div class="cover-empty" style="width: 3%"></div></div>
	</td>
	<td data-value="97.01" class="pct high">97.01%</td>
	<td data-value="67" class="abs high">65/67</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="97.01" class="pct high">97.01%</td>
	<td data-value="67" class="abs high">65/67</td>
	</tr>

<tr>
	<td class="file high" data-value="parseUsp2006.js"><a href="parseUsp2006.js.html">parseUsp2006.js</a></td>
	<td data-value="84.21" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 84%"></div><div class="cover-empty" style="width: 16%"></div></div>
	</td>
	<td data-value="84.21" class="pct high">84.21%</td>
	<td data-value="57" class="abs high">48/57</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="84.21" class="pct high">84.21%</td>
	<td data-value="57" class="abs high">48/57</td>
	</tr>

<tr>
	<td class="file high" data-value="parsers.js"><a href="parsers.js.html">parsers.js</a></td>
	<td data-value="82.6" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 82%"></div><div class="cover-empty" style="width: 18%"></div></div>
	</td>
	<td data-value="82.6" class="pct high">82.6%</td>
	<td data-value="23" class="abs high">19/23</td>
	<td data-value="42.85" class="pct low">42.85%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="82.6" class="pct high">82.6%</td>
	<td data-value="23" class="abs high">19/23</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-02-21T14:54:35.389Z
            </div>
        <script src="../../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../../sorter.js"></script>
        <script src="../../../../../block-navigation.js"></script>
    </body>
</html>
    