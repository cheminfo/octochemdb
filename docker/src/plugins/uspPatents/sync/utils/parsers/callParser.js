import { open } from 'fs/promises';

import { parseStream } from 'arraybuffer-xml-parser';

import { parseUsp2006 } from './parseUsp2006.js';

async function doAll() {
  const xmlPath = '../__tests__/data/2006.xml';
  const fileStream = await open(xmlPath, 'r');
  const readableStream = fileStream.readableWebStream();
  let count = 0;
  let results;
  for await (const entry of parseStream(
    readableStream,
    'patent-application-publication',
  )) {
    results = parseUsp2006(entry);
    count++;
    if (!results) {
      break; // shut to avoid eslint error, this function is used for development only and will be erased soon
    }
    if (count > 0) {
      break;
    }
  }
}

doAll();
