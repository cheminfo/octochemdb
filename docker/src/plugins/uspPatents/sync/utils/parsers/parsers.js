// this function should switch parser function by second argument (parserYear)
import { parse01To04 } from './parsersFrom01To23/parse01To04.js';
import { parse05 } from './parsersFrom01To23/parse05.js';
import { parse06 } from './parsersFrom01To23/parse06.js';
import { parse07To21 } from './parsersFrom01To23/parse07To21.js';
import { parse22a } from './parsersFrom01To23/parse22a.js';
import { parse22bTo23 } from './parsersFrom01To23/parse22bTo23.js';

export function parsers(entry, parserYear) {
  switch (parserYear) {
    case '2001':
      return parse01To04(entry);
    case '2002':
      return parse01To04(entry);
    case '2003':
      return parse01To04(entry);
    case '2004':
      return parse01To04(entry);
    case '2005':
      return parse05(entry);
    case '2006':
      return parse06(entry);
    case '2007':
      return parse07To21(entry);
    case '2008':
      return parse07To21(entry);
    case '2009':
      return parse07To21(entry);
    case '2010':
      return parse07To21(entry);
    case '2011':
      return parse07To21(entry);
    case '2012':
      return parse07To21(entry);
    case '2013':
      return parse07To21(entry);
    case '2014':
      return parse07To21(entry);
    case '2015':
      return parse07To21(entry);
    case '2016':
      return parse07To21(entry);
    case '2017':
      return parse07To21(entry);
    case '2018':
      return parse07To21(entry);
    case '2019':
      return parse07To21(entry);
    case '2020':
      return parse07To21(entry);
    case '2021':
      return parse07To21(entry);
    case '2022a':
      return parse22a(entry);
    case '2022b':
      return parse22bTo23(entry);
    case '2023':
      return parse22bTo23(entry);
    default:
      return null;
  }
}
