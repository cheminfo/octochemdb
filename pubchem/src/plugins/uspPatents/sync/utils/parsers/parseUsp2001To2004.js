/*
    2001 version 1.5 tag:'patent-application-publication'
    2002,2003,2004 version 1.6
    2005 XML Version 4.0 ICE
    2006 XML Version 4.1 ICE
    2007,2008,2009,2010,2011,2012 XML Version 4.2 ICE
    2013,2014 XML Version 4.3 ICE
    2015,2016,2017,2018,2019,2020,2021 XML Version 4.4 ICE
    2022 XML Version 4.5 or 4.6 ICE
    */
export function parseUsp2001To2004(entry) {
  let usp = {};
  // Get document reference id
  const docRef = entry['subdoc-bibliographic-information']['document-id'];
  usp.id = `US-${docRef['doc-number']}-${docRef['kind-code']}`;
  usp.pubchemPatentId = `US${docRef['doc-number']}${docRef['kind-code']}`;
  usp.title =
    entry['subdoc-bibliographic-information']['technical-information'][
      'title-of-invention'
    ];
  usp.abstract = entry['subdoc-abstract'].paragraph['#text'];
  usp.applicationType =
    entry['subdoc-bibliographic-information']['publication-filing-type'];
  usp.dateProduced =
    entry['subdoc-bibliographic-information']['domestic-filing-data'][
      'filing-date'
    ];
  usp.datePublished =
    entry['subdoc-bibliographic-information']['document-id']['document-date'];

  let results = {
    _id: usp.id,
    data: {},
  };
  if (usp.title) {
    results.data.title = usp.title;
  }
  if (usp.abstract) {
    results.data.abstract = usp.abstract;
  }

  if (usp.dateProduced) {
    results.data.dateProduced = usp.dateProduced;
  }
  if (usp.datePublished) {
    results.data.datePublished = usp.datePublished;
  }
  if (usp.applicationType) {
    results.data.applicationType = usp.applicationType;
  }
  if (usp.patentNumber) {
    results.data.patentNumber = usp.patentNumber;
  }
  if (usp.pubchemPatentId) {
    results.data.pubchemPatentId = usp.pubchemPatentId;
  }

  return results;
}
