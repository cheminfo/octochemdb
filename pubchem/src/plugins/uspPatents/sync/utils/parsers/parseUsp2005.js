/*
    2001 version 1.5 tag:'patent-application-publication'
    2002,2003,2004 version 1.6
    2005 XML Version 4.0 ICE
    2006 XML Version 4.1 ICE
    2007,2008,2009,2010,2011,2012 XML Version 4.2 ICE
    2013,2014 XML Version 4.3 ICE
    2015,2016,2017,2018,2019,2020,2021 XML Version 4.4 ICE
    2022 XML Version 4.5 or 4.6 ICE
    */
export function parseUsp2005(entry) {
  let usp = {};
  const documentReference =
    entry['us-bibliographic-data-application']['publication-reference'];
  usp.id = `${documentReference['document-id'].country}-${documentReference['document-id']['doc-number']}-${documentReference['document-id'].kind}`;
  usp.pubchemPatentId = `${documentReference['document-id'].country}${documentReference['document-id']['doc-number']}${documentReference['document-id'].kind}`;
  usp.title =
    entry['us-bibliographic-data-application']['invention-title']['#text'];
  usp.abstract = entry.abstract.p['#text'];
  usp.applicationType =
    entry['us-bibliographic-data-application']['application-reference'][
      '$appl-type'
    ];
  usp.dateProduced = entry['$date-produced'];
  usp.datePublished = entry['$date-publ'];
  usp.country = entry.$country;
  usp.status = entry.$status;
  usp.language = entry.$lang;
  // define final result
  let results = {
    _id: usp.id,
    data: {},
  };

  if (usp.title) {
    results.data.title = usp.title;
  }
  if (usp.abstract) {
    results.data.abstract = usp.abstract;
  }
  if (usp.status) {
    results.data.status = usp.status;
  }

  if (usp.country) {
    results.data.country = usp.country;
  }
  if (usp.language) {
    results.data.language = usp.language;
  }
  if (usp.dateProduced) {
    results.data.dateProduced = usp.dateProduced;
  }
  if (usp.datePublished) {
    results.data.datePublished = usp.datePublished;
  }
  if (usp.applicationType) {
    results.data.applicationType = usp.applicationType;
  }
  if (usp.patentNumber) {
    results.data.patentNumber = usp.patentNumber;
  }
  if (usp.pubchemPatentId) {
    results.data.pubchemPatentId = usp.pubchemPatentId;
  }
  return results;
}
