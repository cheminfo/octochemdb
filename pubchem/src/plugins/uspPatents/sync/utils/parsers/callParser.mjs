import { open } from 'fs/promises';

import { parseStream } from 'arraybuffer-xml-parser';

import { parseUsp2005 } from './parseUsp2005.js';

async function doAll() {
  const xmlPath = '../__tests__/data/2005.xml';
  const fileStream = await open(xmlPath, 'r');
  const readableStream = fileStream.readableWebStream();
  let count = 0;
  let results;
  for await (const entry of parseStream(
    readableStream,
    'us-patent-application',
  )) {
    results = parseUsp2005(entry);
    count++;
    if (count > 0) {
      break;
    }
  }
}

doAll();
