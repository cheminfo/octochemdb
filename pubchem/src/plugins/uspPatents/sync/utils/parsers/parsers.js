// this function should switch parser function by second argument (parserYear)
import { parseUsp2001To2004 } from './parseUsp2001To2004.js';
import { parseUsp2005 } from './parseUsp2005.js';

export function parsers(entry, parserYear) {
  switch (parserYear) {
    case '2001':
      return parseUsp2001To2004(entry);
    case '2002':
      return parseUsp2001To2004(entry);
    case '2003':
      return parseUsp2001To2004(entry);
    case '2004':
      return parseUsp2001To2004(entry);
    case '2005':
      return parseUsp2005(entry);
    default:
      return null;
  }
}
