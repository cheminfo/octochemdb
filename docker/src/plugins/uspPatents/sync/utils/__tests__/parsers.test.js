import { open } from 'fs/promises';
import { join } from 'path';

import { parseStream } from 'arraybuffer-xml-parser';

import { parsers } from '../parsers/parsers.js';

describe('parsers', () => {
  it('2002', async () => {
    const xmlPath = join(__dirname, 'data/2002.xml');
    const fileStream = await open(xmlPath, 'r');
    const readableStream = fileStream.readableWebStream();
    let count = 0;
    let results;
    for await (const entry of parseStream(
      readableStream,
      'patent-application-publication',
    )) {
      results = parsers(entry, '2002');
      count++;
      if (count > 0) {
        break;
      }
    }
    await fileStream.close();
    expect(results).toMatchSnapshot();
  });
  it('2005', async () => {
    const xmlPath = join(__dirname, 'data/2005.xml');
    const fileStream = await open(xmlPath, 'r');
    const readableStream = fileStream.readableWebStream();
    let count = 0;
    let results;
    for await (const entry of parseStream(
      readableStream,
      'us-patent-application',
    )) {
      results = parsers(entry, '2005');
      count++;
      if (count > 0) {
        break;
      }
    }
    await fileStream.close();
    expect(results).toMatchSnapshot();
  });
  it('2006', async () => {
    const xmlPath = join(__dirname, 'data/2006.xml');
    const fileStream = await open(xmlPath, 'r');
    const readableStream = fileStream.readableWebStream();
    let count = 0;
    let results;
    for await (const entry of parseStream(
      readableStream,
      'patent-application-publication',
    )) {
      results = parsers(entry, '2006');
      count++;
      if (count > 0) {
        break;
      }
    }
    await fileStream.close();
    expect(results).toMatchSnapshot();
  });
});
