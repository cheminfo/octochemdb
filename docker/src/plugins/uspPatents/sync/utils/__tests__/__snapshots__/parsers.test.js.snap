// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parsers > 2002 1`] = `
{
  "_id": "US-20020000001-A1",
  "data": {
    "abstract": "A headwear assembly that includes a reversible crown, shaped to fit the head of a wearer with a prong buckle, fabric with hook and loop fasteners, fabric ties, plastic, one-size-fits-all or other size adjustment mechanisms, having first and second opposing major surfaces and a detachable and reversible single visor with first and second opposing major surfaces. A detachable and reversible dual visor with first and second opposing major surfaces can also be attached to the reversible crown. The reversible crown includes cooperating attachment members located on the inside non-focal surfaces of the first and second major opposing surfaces, providing the base for attaching, detaching and reattaching the removable and reversible single or dual visor. The detachable and reversible single visor has first and second major opposing surfaces along with cooperating attachment members to used to attach, detach and reattach itself to the reversible crown. The dual visor also has first and second major opposing surfaces with cooperating attachment members to attach, detach and reattach to the reversible crown. The visor and crown are easily removable and reversible to include one major surface each being displayed upwardly during wear and accounts for many different headwear assembly looks as well as having the reversible crown worn alone.",
    "applicationType": "new",
    "dateProduced": 20010418,
    "datePublished": 20020103,
    "pubchemPatentId": "US20020000001A1",
    "title": "Headwear assembly with reversible crown and detachable, reversible single or dual visor",
  },
}
`;

exports[`parsers > 2005 1`] = `
{
  "_id": "US-20050000001-A1",
  "data": {
    "abstract": "Novelty play jeans for children which include one or more active play features usable while the jeans are being worn. These various features include, but are not limited to a race course for a toy race car, a concealed pocket extractable to form a flag-like decoration, a window pocket, a secret tunnel pocket, a carry pouch convertible into a flat play area, and concealed knee pockets which enable the user of jeans to display a puppet show using built-in features of the jeans, while worn by the user.",
    "applicationType": "utility",
    "country": "US",
    "dateProduced": 20041222,
    "datePublished": 20050106,
    "language": "EN",
    "pubchemPatentId": "US20050000001A1",
    "status": "PARALLEL-RUN",
    "title": "Novelty jeans",
  },
}
`;

exports[`parsers > 2006 1`] = `
{
  "_id": "US-20030000001-A1",
  "data": {
    "abstract": "A multi-phase headset () for aircraft crew members is provided which includes an inflatable gas mask unit () and a visor unit (), supported on the head of a user () by mounting assembly (). The overall headset () also has a selectively and/or automatically operable motive and control assembly (). Preferably; the mounting assembly () includes a pair of ear pieces () and a strap assembly () to allow positioning of the headset () on the user&apos;s head, with the mask and visor units () adjacent the crown of the user&apos;s head. In the event of an emergency, the mask unit () is lowered and the mask body () is inflated for delivery of breathable gas to the user (). Also, the visor unit () may be lowered to a use position atop mask unit (). The pneumatically or electrically operated motive and control assembly () serves to move the mask and visor units () between their retracted and deployed position. In order to prevent ingress of smoke, physical seals or air curtain passageways () may be provided along the upper and lower margins of the visor and mask units ().",
    "dateProduced": 20020524,
    "datePublished": 20030102,
    "patentNumber": 20030000001,
    "pubchemPatentId": "US20030000001A1",
  },
}
`;
